src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -110,10 +111,28 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
           this.props.onSelectedDataSources([{ id: optionId, label: selectedDataSource.title }]);
         }
       } catch (error) {
-        handleDataSourceFetchError(
+        // since we do not set the defaultDataSource when hit error, we need to set it before moving forward
+        if (defaultDataSource) {
+          const { dataSourceFilter } = this.props;
+          // check if the defaultDataSouce can be filtered out
+          const filteredDataSource = dataSourceFilter
+            ? [{ id: defaultDataSource }].filter(dataSourceFilter)
+            : [{ id: defaultDataSource }];
+          const filteredDefaultDataSource =
+            filteredDataSource.length === 0 ? null : filteredDataSource[0].id;
+          this.setState({
+            defaultDataSource: filteredDefaultDataSource,
```

Comment:
When hit error, we get the defaultDataSource from uiSettings, but never set it to local state, if we want to use it out of the componentDidMount, we cannot use it due to the block scope, therefore, I check if it exist and set it to the local state, so that we can use it in other handlers